packages/language/src/language-server/text-documents.ts

```diff
@@ -208,8 +208,28 @@ export class NormalizedTextDocuments<T extends { uri: string }>
     return this._onDidClose.event;
   }
 
-  public get(uri: string | URI): T | undefined {
-    return this._syncedDocuments.get(UriUtils.normalize(uri));
+  public get(
+    uri: string | URI,
+    options?: { loadFromURI?: boolean },
+  ): T | undefined {
+    let syncedDocument = this._syncedDocuments.get(UriUtils.normalize(uri));
+    if (syncedDocument === undefined && options?.loadFromURI) {
+      try {
+        const uriName = typeof uri === "string" ? uri : uri.toString();
```

Comment:
Suggestion: We are already calling `UriUtils.normalize`. This is exactly for creating a string from an `string | URI` typed variable. You should reuse the result from that instead of doing `uri.toString()` again.